File: data_structures/arrays_and_strings/space_replacer.rb
def space_replacer(str)
  chars = str.strip.split('')
  chars.each_with_index do |el, idx|
    chars[idx] = '%20' if el == ' '
  end
  chars.join('')
end

# Time complexity: O(n)
# Space complexity: O(n)
# Note the strip, which gets rid of the trailing whitespace

File: data_structures/arrays_and_strings/space_replacer.js
function spaceReplacer(str) {
  let chars = str.trim().split("");
  for (let i = 0; i < chars.length; i++) {
    if (chars[i] === " ") {
      chars[i] = "%20";
    }
  }
  return chars.join("");
}

// Initial mistakes: Forgot to call trim by adding ()